### PR TITLE
feat: builtin resource definitions

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**
Add builtin resource definitions for kubernetes and docker.

**Related Issue:**
#1708

### CODE REVIEW

File: pkg/resourcedefinitions/builtin.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/seal-io/walrus/pkg/dao"
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/model/resourcedefinition"
+	"github.com/seal-io/walrus/pkg/dao/model/templateversion"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/utils/log"
+)
+
+var (
+	logger = log.WithName("builtin").WithName("resource-definitions")
```
Comment (thxCode) [inline pkg/resourcedefinitions/builtin.go:18]: keep the logger in function scope

File: pkg/resourcedefinitions/builtin.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	ctx context.Context,
+	mc model.ClientSet,
+	resourceType string,
+	connectorType string,
+) (*model.ResourceDefinitionMatchingRule, error) {
+	name := fmt.Sprintf("%s-%s", connectorType, resourceType)
+
+	version, err := mc.TemplateVersions().Query().
+		Where(templateversion.Name(name)).
+		Order(model.Desc(templateversion.FieldVersion)).
```
Comment (thxCode) [inline pkg/resourcedefinitions/builtin.go:97]: we should compare the semver versions here?

File: pkg/resourcedefinitions/builtin.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/seal-io/walrus/pkg/dao/model/templateversion"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/utils/log"
+)
+
+var (
+	logger = log.WithName("builtin").WithName("resource-definitions")
+
+	resourceTypes  = []string{"containerservice", "mysql", "postgresql", "redis"}
+	connectorTypes = []string{types.ConnectorTypeKubernetes, types.ConnectorTypeDocker}
```
Comment (thxCode) [inline pkg/resourcedefinitions/builtin.go:21]: work with repo topics

File: pkg/resourcedefinitions/builtin.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		Selector: types.Selector{EnvironmentLabels: map[string]string{
+			dao.ProviderLabelPrefix + connectorType: dao.LabelValueTrue,
+		}},
+		Edges: model.ResourceDefinitionMatchingRuleEdges{
+			Template: version,
+		},
+	}
+
+	if connectorType == types.ConnectorTypeDocker {
+		v, err := json.Marshal(map[string]string{"network_id": "local-walrus"})
```
Comment (gitlawr) [inline pkg/resourcedefinitions/builtin.go:151]: Do you think we can use this as the default value in docker templates?

Comment (shanewxy) [reply pkg/resourcedefinitions/builtin.go]: Yes. I'll modify the templates schema and remove this.

**Comment by gitlawr** [comment]
Can we provide a way to opt out of creating the builtin definitions? In case when users want to define their own `mysql`, `redis`, etc.

**Comment by shanewxy** [comment]
> Can we provide a way to opt out of creating the builtin definitions? In case when users want to define their own `mysql`, `redis`, etc.

Updated. Added a setting for builtin resource definitions.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by thxCode — APPROVED** [review]
LGTM

**Pull request merged**